<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 3.2//EN">
<html>
  <head>
<title>GEOSS Home Page</title>
<meta name="keywords" content="gene, chip, expression, affymetrix, genechip, oligo, two, color, spotted, array, dna, rna, oligonucleotide, cdna, analysis, analytical, copy, copy dna, spotted array, hybridization, mrna, messenger, messenger rna, cy3, cy5, probe, reverse transcriptase, reverse transcription, glass, slide, sage, serial, serial analysis of gene expression, photo dimerization, genotype, genotypes, genotyping, fold change, over expression, under expression, up regulation, down regulation, photo lithography, linux, open source, apache, postgres, postgresql, database, secure, curate, curation, curator, perfect match, westfa, young, stat analysis, clustering, perm cluster, r-cluster, r cluster, tree view, go, gene ontology, flurochroms, patrick brown, t-test, anova, function prediction">
<meta name="description" content="GEOSS is an Open Source Gene Expression Database supporting storage and analysis of Affymetrix&reg; GeneChip&reg; technology, with planned support for two color spotted arrays and other gene expression measurement technologies. GEOSS is designed to serve as an secure repository and archive for many researchers' data. ">
</head>

  
<body bgcolor="#FFFFFF" marginheight="0" marginwidth="0" topmargin="0"
  leftmargin="0" link="#006633" vlink="#666633">
<table width="600" border="0" cellspacing="0" cellpadding="0">
  <tr> 
    <td width="8" valign="top">&nbsp;</td>
    <td width="10" bgcolor="990066" valign="bottom"><img src="graphics/white.gif" width="10" height="10"></td>
    <td width="10" valign="bottom"><img src="graphics/white.gif" width="10" height="10"></td>
    <td width="580"> 
      <table width="600" border="0" cellspacing="0" cellpadding="0">
        <tr valign="top"> 
          <td width="100"><font face="Verdana, Arial, Helvetica, sans-serif" size="+1"><a href="./index.cgi"><img src="graphics/geoss_logo.jpg" width="216" height="96" border="0"></a></font></td>
          <td width="8">&nbsp;&nbsp;</td>
          <td width="490"><font face="Verdana, Arial, Helvetica, sans-serif" size="+1">GEOSS: 
            Microarray Analysis and Archive<br>
            </font><font face="Verdana, Arial, Helvetica, sans-serif">Created 
            at the University of Virginia</font> </td>
        </tr>
      </table>
      <p><b>What is GEOSS?</b><br>
        GEOSS is a repository and data analysis suite for gene microarray data. 
        The acronym stands for Gene Expression Open Source System.</p>
      <p><b>Researchers at the University of Virginia</b><br>
        If you are performing microarray studies with the BRF at UVA, you'll need 
        an account on the UVA GEOSS system. Please email <a href="mailto:geoss-help@virginia.edu">geoss-help@virginia.edu</a> 
        for an account. To see learn about using GEOSS, please see our <a href="stepbystep.html">Step 
        By Step Guide</a>.</p>
      <p><b>Downloading GEOSS</b><br>
        The GEOSS package is also available as a download. With its easy to 
        use web interface, strong security and ever expanding analysis capabilities, 
        GEOSS is an ideal way to manage microarray data. It also includes features 
        to help your microarray center manage the chip workflow. GEOSS is a 
        mature, open source package freely available. Running GEOSS requires 
        a Linux server. Installation takes several hours. Extensive documentation 
        is available for administrators and end users. The GEOSS developers 
        are available for questions via email. For more details about downloading, 
        see the <a href="download.html">download page</a>.</p>
      <p><b>From the Researcher's Point of View</b><br>
        We built GEOSS to support genome-wide microarray studies at the University 
        of Virginia. Base features are data repository, database, Analysis Tree 
        analysis, security, and microarray center workflow and chip tracking.</p>
      <p>From the investigator's point of view, GEOSS is the place to create 
        orders for genechip microarray assays, retrieve data for completed assays, 
        and analyze data. If the time comes when the data needs to be made publicly 
        accessible for purposes of publication, GEOSS also has a MIAME complaint 
        public data repository.</p>
      <p>If you have technical questions regarding web access and/or your UVA 
        GEOSS account.cgiease send an email to <a href="mailto:geoss-help@virginia.edu">geoss-help@virginia.edu</a>.</p>
      <p>If you have general questions about your array study, such as RNA sample 
        preparation, chip selection, and costs, you can address them to:</p>
      <p>Alyson Prorock, <a href="mailto:ajp7x@virginia.edu">ajp7x@virginia.edu</a> 
        3-9689<br>
        Yongde Bao, Ph.D., Manager of Microarray Facility <a href="mailto:yb8d@virginia.edu">yb8d@virginia.edu</a> 
        2-2551<br>
        Jay Fox, Ph.D., BRF Director <a href="mailto:jwf8x@virginia.edu">jwf8x@virginia.edu</a> 
        4-0050</p>
      <p> <strong>GMB array data analysis service <br>
        </strong>In addition to the many features of the GEOSS database and 
        analysis system, the GMB Core offers a formal analysis service to University 
        of Virginia array users. This analysis service will provide the results 
        of quality control, statistical evaluation and discovery of differential 
        expression, clustering &amp; classification analysis, and some annotation 
        analysis for each array study. This service is partially supported by 
        the UVA School of Medicine, but some fees will be assessed starting in 
        January 1, 2003. The BRF will continue provide basic analysis of each 
        array data set. Questions regarding the array analysis service can be 
        addressed to </p>
      <p>Mir Siadaty, M.S. <a href="mailto:mss4x@virginia.edu">mss4x@virginia.edu</a> 
        2-4436<br>
        Jae Lee, Ph.D., GMB Director <a href="mailto:jkl7n@virginia.edu">jkl7n@virginia.edu</a> 
        2-1033</p>
      <p><br>
        <br>
      </p>
    </td>
  </tr>
</table>
<p>&nbsp;</p>
<p>&nbsp;</p>
</body>
</html>


